### PR TITLE
fix(composite services): Don't use decision returned from inner service when it returns an error

### PR DESCRIPTION
**Pull request by mjc1283**
Summary:

- In `CompositeExperimentService` and `CompositeFeatureService` `GetDecision` methods, when the inner service returns an error, skip that decision and move to the next inner service
- Return an error from `CompositeExperimentService` `GetDecision` when no decision was made

Test Plan:

Updated & added unit tests

### CODE REVIEW

File: pkg/decision/composite_experiment_service.go [hunk 1]
```diff
@@ -47,10 +53,14 @@ func (s CompositeExperimentService) GetDecision(decisionContext ExperimentDecisi
...
+		if err != nil {
+			ceLogger.Debug(fmt.Sprintf("%v", err))
+		}
+		if decision.Variation != nil && err == nil {
 			return decision, err
 		}
 	}
 
-	return experimentDecision, nil
+	return experimentDecision, errors.New("no decision was made")
```
Comment (mikeng13) [inline pkg/decision/composite_experiment_service.go]: Should we log this error or should the parent `composite_service` log it?

File: pkg/decision/composite_feature_service.go [hunk 2]
```diff
@@ -45,10 +46,13 @@ func NewCompositeFeatureService(compositeExperimentService ExperimentService) *C
...
+		if err != nil {
+			cfLogger.Debug(fmt.Sprintf("%v", err))
+		}
+		if featureDecision.Variation != nil && err == nil {
 			return featureDecision, err
 		}
 	}
 
-	return FeatureDecision{}, fmt.Errorf("no decision was made for feature %s", decisionContext.Feature.Key)
+	return FeatureDecision{}, errors.New("no decision was made")
```
Comment (mikeng13) [inline pkg/decision/composite_feature_service.go]: Same thing re: logging

File: pkg/decision/composite_experiment_service.go (hunk 1 above)
Comment (mjc1283) [reply pkg/decision/composite_experiment_service.go]: If I'm reading correctly, the client will end up logging this error, and the same is true for the composite feature service.

Comment (mikeng13) [reply pkg/decision/composite_experiment_service.go]: Ah got it, ship away then!

**Pull request merged**